collections/src/main/scala/strawman/collection/mutable/WrappedArray.scala

```diff
@@ -0,0 +1,242 @@
+package strawman.collection
+package mutable
+
+import scala.{Unit, Int, Array, Boolean, Any, Byte, Short, Char, Long, Float, Double, AnyRef, Serializable, specialized, `inline`}
+import scala.Predef.{Class, implicitly}
+import scala.runtime.ScalaRunTime
+import scala.reflect.ClassTag
+import scala.util.hashing.MurmurHash3
+
+import java.util.Arrays
+
+/**
+  *  A collection representing `Array[T]`. Unlike `ArrayBuffer` it is always backed by the same
+  *  underlying `Array`, therefore it is not growable or shrinkable.
+  *
+  *  @tparam T    type of the elements in this wrapped array.
+  *
+  *  @author  Martin Odersky, Stephane Micheloud
+  *  @version 1.0
+  *  @since 2.8
+  *  @define Coll `WrappedArray`
+  *  @define coll wrapped array
+  *  @define orderDependent
+  *  @define orderDependentFold
+  *  @define mayNotTerminateInf
+  *  @define willNotTerminateInf
+  */
+abstract class WrappedArray[T]
+  extends AbstractSeq[T]
+    with IndexedSeq[T]
+    with IndexedSeqOps[T, WrappedArray, WrappedArray[T]]
+    with ArrayLike[T]
+    with IndexedOptimizedSeq[T]
+    with StrictOptimizedSeqOps[T, WrappedArray, WrappedArray[T]]
+    with Serializable {
+
+  def iterableFactory: strawman.collection.SeqFactory[WrappedArray] = WrappedArray
+
+  protected[this] def fromSpecificIterable(coll: strawman.collection.Iterable[T]): WrappedArray[T] = {
+    val b = ArrayBuilder.make()(elemTag)
+    val s = coll.knownSize
+    if(s > 0) b.sizeHint(s)
+    b ++= coll
+    WrappedArray.make(b.result())
+  }
```

Comment:
Looks like `iterableFactory.from(coll)` would be a better implementation.

Reply:
No, the `IterableFactory` cannot make use of the existing `elemTag`. It would always have to build a boxed array.